File: src/disaggregated_component/memory_module/MemoryModule.py
from m5.params import *
from m5.proxy import *
from m5.objects.BaseModule import BaseModule


class MemoryModule(BaseModule):
    type = 'MemoryModule'
    cxx_header = "disaggregated_component/memory_module/memory_module.hh"
    is_dummy = True

File: src/disaggregated_component/memory_module/memory_module.cc
#include "disaggregated_component/memory_module/memory_module.hh"

MemoryModule::MemoryModule(Params *p):
    BaseModule(p)
{}

MemoryModule*
MemoryModuleParams::create()
{
    return new MemoryModule(this);
}

void
MemoryModule::init()
{
    // removed system port check
    return;
}
